src/xma/src/xmaplugin/xmaplugin.cpp

```diff
@@ -994,7 +985,7 @@ int32_t xma_plg_cu_cmd_status(XmaSession s_handle, XmaCUCmdObj* cmd_obj_array, i
         } else if (!all_done) {
             if (g_xma_singleton->cpu_mode == XMA_CPU_MODE1) {
                 std::unique_lock<std::mutex> lk(priv1->m_mutex);
-                priv1->work_item_done_1plus.wait(lk);
+                priv1->kernel_done_or_free.wait(lk);
```

Comment:
What happens if this condition wakes up spuriously?

Reply:
In that case it will redo all check and go back to wait if required